packages/amplify-codegen-e2e-tests/src/codegen-tests-base/add-codegen.ts

```diff
@@ -29,3 +35,105 @@ export async function testAddCodegen(config: AmplifyFrontendConfig, projectRoot:
+        writeFileSync(path.join(projectRoot, sdlFilename), sdlSchema);
+    }
+
+    // add codegen without init
+    switch (config.frontendType) {
+        case AmplifyFrontend.javascript:
+            await addCodegenNonAmplifyJS(projectRoot, initialFailureMessage);
+            break;
+        case AmplifyFrontend.typescript:
+            await addCodegenNonAmplifyTS(projectRoot, initialFailureMessage);
+            break;
+        default:
+            throw new Error(`Received unexpected frontendType ${config.frontendType}`);
+    }
+
+    // return if we expected the add command to fail
+    if (initialFailureMessage) {
+        return;
+    }
+
+    // pre-existing file should still exist
+    expect(existsSync(userSourceCodePath)).toBe(true);
+    // GraphQL statements are generated
+    const generatedStatementFiles = new Set(readdirSync(path.join(projectRoot, config.graphqlCodegenDir)));
+    console.log(`Comparing written files: ${JSON.stringify(Array.from(generatedStatementFiles))} to expected files: ${JSON.stringify(expectedFilenames)}`);
+    expectedFilenames.forEach((expectedFilename) => expect(generatedStatementFiles.has(expectedFilename)).toBe(true));
+    // graphql configuration should be added
+    expect(existsSync(getGraphQLConfigFilePath(projectRoot))).toBe(true);
+    if (config.frontendType === AmplifyFrontend.typescript) {
+        assertTypeFileExists(projectRoot)
+    }
+
+    if (dropAndRunCodegen || dropAndRunCodegenStatements || dropAndRunCodegenTypes) {
+        rmSync(path.join(projectRoot, config.graphqlCodegenDir), { recursive: true });
+        // pre-existing file should still exist
+        expect(existsSync(userSourceCodePath)).toBe(true);
+        // Graphql statements are deleted
+        expect(existsSync(path.join(projectRoot, config.graphqlCodegenDir))).toBe(false);
+    }
+
+    if (dropAndRunCodegen) {
+        await generateStatementsAndTypes(projectRoot);
+
+        // GraphQL statements are regenerated
+        const regeneratedStatementFiles = new Set(readdirSync(path.join(projectRoot, config.graphqlCodegenDir)));
+        console.log(`Comparing written files: ${JSON.stringify(Array.from(regeneratedStatementFiles))} to expected files: ${JSON.stringify(expectedFilenames)}`);
+        expectedFilenames.forEach((expectedFilename) => expect(regeneratedStatementFiles.has(expectedFilename)).toBe(true));
+
+        if (config.frontendType === AmplifyFrontend.typescript) {
+            assertTypeFileExists(projectRoot)
+        }
+    }
+
+    if (dropAndRunCodegenStatements) {
+        await generateStatements(projectRoot);
+
+        // GraphQL statements are regenerated
+        const regeneratedStatementFiles = new Set(readdirSync(path.join(projectRoot, config.graphqlCodegenDir)));
+        console.log(`Comparing written files: ${JSON.stringify(Array.from(regeneratedStatementFiles))} to expected files: ${JSON.stringify(expectedFilenames)}`);
```

Comment:
+2

Reply:
:D

Reply:
I've pulled this into a utility method to reduce the duplication in this function.